libr/bin/bfile.c

```diff
@@ -847,3 +848,41 @@ R_API bool r_bin_file_close(RBin *bin, int bd) {
 	}
 	return false;
 }
+
+R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
+	char hash[128], *p;
+	RHash *ctx;
+	ut8* buf;
+	ut64 buf_len = 0;
+	int i;
+	RBinFile *bf = bin->cur;
+	RBinObject *o = bf->o;
+	RIODesc *iod = r_io_desc_get (bin->iob.io, bf->fd);
+
+	if (!file && iod) {
+		file = iod->name;
+	}
+
+	buf_len = r_io_desc_size (iod);
+	if (buf_len > limit)
+		buf_len = limit;
+	buf = calloc (1, buf_len);
+	r_io_desc_read (iod, buf, buf_len);
+	ctx = r_hash_new (false, R_HASH_MD5 | R_HASH_SHA1);
+	for (i = 0; i < buf_len; i += 1024) {
+		(void)r_hash_do_md5 (ctx, &buf[i], 1024);
+		(void)r_hash_do_sha1 (ctx, &buf[i], 1024);
+	}
```

Comment:
the last block (if file is not % 1024) is not hashed

Reply:
err i mean it will hash more bytes than the ones available in the buffer, you must R_MIN(1024, buf_len-i)

Reply:
Ah right ...